smart_open/http.py

```diff
@@ -48,7 +48,12 @@ def __init__(self, url, mode='r', buffer_size=DEFAULT_BUFFER_SIZE,
         self.buffer_size = buffer_size
         self.mode = mode
 
-        self.response = requests.get(url, auth=auth, stream=True, headers=_HEADERS)
+        if headers:
+            self.headers = headers
+        else:
+            self.headers = _HEADERS
```

Comment:
Similar objection: we don't want instances (individual objects) to share globals in any way.

To see what I mean, consider this scenario:

```python
x = BufferedInputBase()  # use default _HEADERS
x.headers['Accept-Encoding'] = 'compress, gzip'  # or whatever
x.headers['Other-Header'] = 'value'  # or whatever

y = BufferedInputBase()  # use default again, but…
print(y.headers)  # surprise!
```

Reply:
Nice catch, thanks for the example @piskvorky

Reply:
Perhaps we could do this:

```python
        if headers:
            self.headers = headers
        else:
            self.headers = dict(_HEADERS)
```

A shallow copy should be sufficient to ensure the contents of _HEADERS remain unchanged.

Reply:
Yes, except `if headers is None` is the standard pattern (a user passing `{}` is not the same as default `None`).